engine/src/lnxspec.cpp

```diff
@@ -697,6 +708,8 @@ char *MCS_resolvepath(const char *path)
 			sptr++;
 		strcpy(sptr, newname);
 		delete newname;
+        // SN-2015-06-05: [[ Bug 15432 ]] Calling MCS_resolvepath ensures that
+        //  we will get an absolute path here, if any
 		newname = MCS_resolvepath(fullpath);
```

Comment:
I'm not 100% sure but I think this might make it possible to cause a recursion overflow... If you can create a symlink which references itself indirectly then you'll end up with an infinite loop.